mono/metadata/icall-def.h

```diff
@@ -151,17 +151,17 @@ ICALL(ARGI_3, "IntGetNextArgType",                mono_ArgIterator_IntGetNextArg
 ICALL(ARGI_4, "Setup",                            mono_ArgIterator_Setup)
 
 ICALL_TYPE(ARRAY, "System.Array", ARRAY_1)
-ICALL(ARRAY_1, "ClearInternal",    ves_icall_System_Array_ClearInternal)
+HANDLES(ICALL(ARRAY_1, "ClearInternal",    ves_icall_System_Array_ClearInternal))
 ICALL(ARRAY_3, "CreateInstanceImpl",   ves_icall_System_Array_CreateInstanceImpl)
 ICALL(ARRAY_14, "CreateInstanceImpl64",   ves_icall_System_Array_CreateInstanceImpl64)
 ICALL(ARRAY_4, "FastCopy",         ves_icall_System_Array_FastCopy)
 ICALL(ARRAY_5, "GetGenericValueImpl", ves_icall_System_Array_GetGenericValueImpl)
-ICALL(ARRAY_6, "GetLength",        ves_icall_System_Array_GetLength)
-ICALL(ARRAY_15, "GetLongLength",        ves_icall_System_Array_GetLongLength)
-ICALL(ARRAY_7, "GetLowerBound",    ves_icall_System_Array_GetLowerBound)
-ICALL(ARRAY_8, "GetRank",          ves_icall_System_Array_GetRank)
-ICALL(ARRAY_9, "GetValue",         ves_icall_System_Array_GetValue)
-ICALL(ARRAY_10, "GetValueImpl",     ves_icall_System_Array_GetValueImpl)
+HANDLES(ICALL(ARRAY_6, "GetLength",        ves_icall_System_Array_GetLength))
+HANDLES(ICALL(ARRAY_15, "GetLongLength",   ves_icall_System_Array_GetLongLength))
+HANDLES(ICALL(ARRAY_7, "GetLowerBound",    ves_icall_System_Array_GetLowerBound))
```

Comment:
`GetLength` and `GetLowerBound` are sometimes replaced by the JIT with intrinsics. https://github.com/mono/mono/blob/1ba158386dfd8fbba3c2d8a60da77dc4189c676a/mono/mini/intrinsics.c#L345-L346

Also there's this bit in `method-to-ir.c` which looks like it will skip making a wrapper for `System.Array` icalls... 
https://github.com/mono/mono/blob/1ba158386dfd8fbba3c2d8a60da77dc4189c676a/mono/mini/method-to-ir.c#L7736-L7744

So I'm a little suspicious about adding handles here.  Can we actually get to this slow path, and if so, do the arguments look right?

Reply:
Agreeed slow path might be difficult to reach. I could add reflection-based tests.

Reply:
Direct icalls I believe currently disabled.

Reply:
@jaykrell sorry I highlighted too much code to try to show context and then didn't actually call out the suspicious bits.  It's not the direct calls. I meant the first condition `cmethod->iflags & METHOD_IMPL_ATTRIBUTE_INTERNAL_CALL && `mini_class_is_system_array (cmethod->klass)`  will cause us to skip the third one (`fsig->pinvoke`) where the call to `mono_marshal_get_native_wrapper` is the bit that actually adds the wrapper around the icall invocation. 

So if I'm reading this right, icalls in `System.Array` will never get a wrapper.

Reply:
Also I  don't fully understand this part of `method-to-ir.c`.  I think we first add the wrapper for the icall, and then replace the call instruction with an intrinsic.

Reply:
I think it's probably difficult to write a test that reliably ends up on the slow path (we do a lot to try to make arrays fast), but it's worth doing a local test (maybe some printf debugging or flipping some bits with the debugger to bypass the intrinsification) just to verify that the args on the slow path get the expected wrappers.